src/publisher.ts

```diff
@@ -82,6 +82,7 @@ export class StreamPublisher implements Publisher {
   private scheduled: NodeJS.Immediate | null
   private logger: Logger
   private maxChunkLength: number
+  private isClosed = false
```

Comment:
I honestly do not like "is" properties, I would suggest using only closed, wdyt?